### ISSUE TITLE
Path separators are platform specific

### ISSUE THREAD

**Issue by marinasundstrom** [marinasundstrom/BookStoreScraper#4]
The user encountered this while running on Windows.

It seems like the ```Uri``` class cannot handle URIs with backwards slashes ``\`` which are used in Windows.

The ``Path`` API,  which are used for Path manipulation, conforms the the host platform. So on Windows, it uses ```\```.

This program was developed on a Mac which used ```/``` in filesystem paths.

Question: Is there a better method than using the platform specific ```Path```.

```sh
info: BookScraper.NavigationManager[0]
      Navigated to: http://books.toscrape.com/index.html
info: BookScraper.Processors.DocumentProcessor[0]
      Downloading: http://books.toscrape.com/index.html
info: BookScraper.MyHttpClient[0]
      Downloading: http://books.toscrape.com/index.html
info: BookScraper.Processors.DocumentProcessor[0]
      Processing document
info: BookScraper.Processors.DocumentProcessor[0]
      Found script: http://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js
info: BookScraper.Processors.DocumentProcessor[0]
      Found script: static/oscar/js/bootstrap3/bootstrap.min.js
Unhandled exception. System.UriFormatException: Invalid URI: The hostname could not be parsed.
   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind, UriCreationOptions& creationOptions)
   at System.Uri..ctor(String uriString)
   at BookScraper.UrlHelpers.AsAbsoluteUrl(String baseUrl, String currentUrl, String relUrl) in C:\code\assignments\marinasundstrom\BookScraper\UrlHelpers.cs:line 15
   at BookScraper.Context.AsAbsoluteUrl(String relUrl) in C:\code\assignments\marinasundstrom\BookScraper\Context.cs:line 17
   at BookScraper.Processors.ScriptProcessor.ProcessScriptSrc(Context context, String scriptElementSrc, CancellationToken cancellationToken) in C:\code\assignments\marinasundstrom\BookScraper\Processors\ScriptProcessor.cs:line 26
   at BookScraper.Processors.DocumentProcessor.ProcessScripts(Context context, IDocument document, CancellationToken cancellationToken) in C:\code\assignments\marinasundstrom\BookScraper\Processors\DocumentProcessor.cs:line 124
   at BookScraper.Processors.DocumentProcessor.ProcessHtml(Context context, IDocument document, CancellationToken cancellationToken) in C:\code\assignments\marinasundstrom\BookScraper\Processors\DocumentProcessor.cs:line 102
   at BookScraper.Processors.DocumentProcessor.ScrapeDocument(Context context, CancellationToken cancellationToken) in C:\code\assignments\marinasundstrom\BookScraper\Processors\DocumentProcessor.cs:line 95
   at BookScraper.Processors.DocumentProcessor.ScrapeDocument(String url, CancellationToken cancellationToken) in C:\code\assignments\marinasundstrom\BookScraper\Processors\DocumentProcessor.cs:line 45
   at BookScraper.Scraper.Scrape(CancellationToken cancellationToken) in C:\code\assignments\marinasundstrom\BookScraper\Scraper.cs:line 64
   at BookScraper.ScraperHostedService.StartAsync(CancellationToken cancellationToken) in C:\code\assignments\marinasundstrom\BookScraper\ScraperHostedService.cs:line 15
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Program.<Main>$(String[] args) in C:\code\assignments\marinasundstrom\BookScraper\Program.cs:line 26
   at Program.<Main>(String[] args)
```

**Comment by marinasundstrom** [comment]
Browsers allow for mixed mode ```/``` and ```\``` but the ```Uri``` class doesn't.

The bug is occurs in ```ToAbsoluteUrl```. Just make sures it outputs a well formatted URL the rest of the program will work.

The fix is to normalize URL by replacing the ```\``` with ```/``` in the ```ToAbsoluteUrl``` method.

**Comment by marinasundstrom** [comment]
This has been fixed.
